src/components/10-atoms/button/index.js

```diff
@@ -64,90 +56,25 @@ class AXAButton extends InlineStyles {
     return this.icon === ARROW_RIGHT;
   }
 
-  watch(mode) {
-    switch (mode) {
-      case 'stop':
-        if (this._observer) {
-          this._observer.disconnect();
-        }
-        break;
-      case 'start':
-        if (!this._observer) {
-          this._observer = new MutationObserver(() => this.attachFakeButton());
-        }
-        this._observer.observe(this, {
-          childList: true,
-        });
-        break;
-      default:
-        break;
+  handleClick(ev) {
+    if (this.type !== 'button' && hasShadowDom(this)) {
```

Comment:
You have 3 special cases, that could be handled by **3 early returns**, then it's more readable and less nested:
- type=button
- no shadowDOM
- no form